sinatra-contrib/lib/sinatra/contrib/setup.rb

```diff
@@ -9,16 +8,15 @@ def extensions
         @extensions ||= {:helpers => [], :register => []}
       end
 
-      def register(name, path = nil)
+      def register(name, path)
```

Comment:
Isn't this a breaking change since `path` is now required for "register", "helpers", and "autoload"?

Reply:
It's been quite a while since I've looked at this so I'm kind of scratching my head here, but AFAICT these methods are local to Sinatra::Contrib and only used to set up the `@extensions` data structure before delegating to the proper Sinatra.autoload method. They don't override the methods defined in Sinatra::Base. So if some third party library is adding itself to the Sinatra::Contrib namespace and expecting these methods to work this way (which wouldn't work anyway, I don't think, because the old method assumes the required files are under sinatra-contrib), those might break, but everything else should be fine.

Reply:
To be fully backwards-compatible we'll have to implement our own String#underscore method, which shouldn't be too difficult (I have the snippet floating around here somewhere). Just let me know what you want to do.

Reply:
Maybe we don't have to worry about it if no one is using it, but since it technically is public we have to do some digging to find out. Let's try with a GitHub search to see if any usage turns up.

Reply:
I have investigated the impact of this change on the outside of sinatra.
As @zzak says, I used a search on github and used rubygems' reverse dependency.
As a result, I couldn't find gem etc which depended on this feature.

Although the investigation doesn't mean that this change has no influence at all, it certainly doesn't at least have a big influence.
If a troubled person appears, we should support it separately by the sinatra community.

Reply:
Even if there is no visible impact we can find on public GitHub repos, introducing this change in a point release breaks semver. I would rather wait and merge it in 2.1, but that is just my 2 cents. 

Admittedly the value of dropping the `backports` dependency is likely higher than a breaking change that _may_ impact no one.

My worry is that someone may be depending on this API in a private extension to their app without knowing, then when they upgrade to 2.0.2 :bomb:

Reply:
Okay, I'll remark this as v2.1.0.

Reply:
According to semver, doesn’t that need to be a major bump in version? https://semver.org/